app/models/supported_ssg.rb

```diff
@@ -89,9 +88,13 @@ def os_version(rhel)
       rhel.scan(/(\d+)\.(\d+)$/)[0]
     end
 
+    def fetch_config
+      ConfigDownloader.ssg_ds_config
+    end
+
     def raw_supported
       # cached for the whole runtime
-      @raw_supported ||= YAML.load_file(self::SUPPORTED_FILE)
+      @raw_supported ||= YAML.safe_load(fetch_config)
```

Comment:
I don't see why not. I also don't see what benefit it'd give us. In an initializer, if something goes wrong, it would seem much more difficult to correct (i.e. restart the pod?) than if some API request fails here and simply retries the next time. Note that this `@raw_supported` is cached after the first usage of SupportedSsg (per thread), so it behaves similarly as being in an initializer anyway, but it's a bit lazier.

Reply:
This way the initial load is mapped to a first request that comes in (that uses this), hence the suggestion to initialize it as part of the application. It should not crash in some random place where it would be used for the first time.

Considering the caching, there is no error handling or retry, so a single failure can lead to incorrect supported matrix for the whole duration of pod. We'd likely need to keep the fallback file in sync with what's served by compliance-ssg. Or maybe the fallback can be completely disabled in production, which would require to have compliance-ssg serving it. Still there are sync problems if we'd deploy a new compliance-ssg along with backend.

Reply:
Still not using an initializer, but I did fix the caching issue. Thank you for pointing it out!